modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1197 @@
+        let registers_amount = (length_of_signature + 1) / 2;
+        let regs = self.read_holding_registers(8188, registers_amount)?;
+        let mut signature = to_hex_string(regs);
+        signature.truncate((length_of_signature * 2) as usize);
+        log::info!("Read the signature: {}", signature);
+        Ok(signature)
+    }
+
+    /// For 15 seconds and checks the signature status every 10 ms
+    fn check_signature_status(&self) -> Result<()> {
+        retry::retry(
+            || {
+                let status = self.read_command_status()?;
+                match status {
+                    SignatureStatus::NotInitialized
+                    | SignatureStatus::Idle
+                    | SignatureStatus::SignatureInProgress => Ok(retry::RetryState::KeepTrying),
+                    SignatureStatus::SignatureOK => Ok(retry::RetryState::Done),
+                    error => anyhow::bail!("Error state {error:?}"),
+                }
+            },
+            Duration::from_secs(15),
+        )
+    }
+
+    fn start_transaction(
+        &self,
+        req: generated::types::powermeter::TransactionReq,
+    ) -> Result<TransactionStartResponse> {
+        // We can start transaction only in Idle state
+        let state = self.read_state()?;
+        match state {
+            IskraMaterState::Active
+            | IskraMaterState::Active_after_power_failure
+            | IskraMaterState::Active_after_reset => {
+                log::error!("Unexpected state {state:?}, trying to stop transaction");
+                self.stop_transaction()?;
+            }
+            IskraMaterState::Idle => {}
+            IskraMaterState::Unknown => {
+                log::warn!("Unknown state");
+            }
+        }
+        self.set_time()?;
+        log::info!("Set time");
+        // set algorithm
+        self.write_single_register(7059, 0)?;
+        self.write_metadata(&req.evse_id)?;
+
+        // Finally send start transaction, we are sending 'B'
+        self.write_single_register(7051, 0x4200)?;
+        log::info!("Started transaction.");
+
+        // Wait until the meter is active.
+        retry::retry(
+            || {
+                let state = self.read_state()?;
+                match state {
+                    IskraMaterState::Active => Ok(retry::RetryState::Done),
+                    _ => Ok(retry::RetryState::KeepTrying),
+                }
+            },
+            std::time::Duration::from_secs(1),
+        )?;
+
+        self.check_signature_status()?;
+        let signature = self.read_signature()?;
+        let signed_meter_values = self.read_signed_meter_values()?;
+        Ok(TransactionStartResponse {
+            error: Option::None,
+            signed_meter_value: Some(SignedMeterValue {
+                signed_meter_data: create_ocmf(signed_meter_values, signature),
+                signing_method: "OCMF".to_string(),
+                encoding_method: String::new(),
```

Comment:
If I understand correctly, in "lem_xxxx" meter here https://github.com/EVerest/everest-core/blob/a6e7ad88a753a2829749e1b1dee00ba3a6c6862e/modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp#L89

The signing method is "" and "OCMF" goes to `encoding_method`.

@hikinggrass what is right?

Reply:
The encoding_method describes the format in which the signed meter data was encoded, so in this case this would be "OCMF" and the signing_method can be left empty, so it's exactly like you suggested: needs to be flipped around